src/data/ellpack_page_raw_format.cu

```diff
@@ -17,26 +17,35 @@ class EllpackPageRawFormat : public SparsePageFormat<EllpackPage> {
  public:
   bool Read(EllpackPage* page, dmlc::SeekStream* fi) override {
     auto* impl = page->Impl();
-    impl->Clear();
-    if (!fi->Read(&impl->matrix.n_rows)) return false;
-    return fi->Read(&impl->idx_buffer);
+    fi->Read(&impl->cuts_.cut_values_.HostVector());
+    fi->Read(&impl->cuts_.cut_ptrs_.HostVector());
+    fi->Read(&impl->cuts_.min_vals_.HostVector());
+    fi->Read(&impl->n_rows);
+    fi->Read(&impl->is_dense);
+    fi->Read(&impl->row_stride);
+    if (!fi->Read(&impl->gidx_buffer.HostVector())) {
+      return false;
+    }
+    return true;
   }
 
   bool Read(EllpackPage* page,
             dmlc::SeekStream* fi,
             const std::vector<bst_uint>& sorted_index_set) override {
-    auto* impl = page->Impl();
-    impl->Clear();
-    if (!fi->Read(&impl->matrix.n_rows))  return false;
-    return fi->Read(&page->Impl()->idx_buffer);
+    LOG(FATAL) << "Not implemented";
+    return false;
   }
 
   void Write(const EllpackPage& page, dmlc::Stream* fo) override {
     auto* impl = page.Impl();
-    fo->Write(impl->matrix.n_rows);
-    auto buffer = impl->idx_buffer;
-    CHECK(!buffer.empty());
-    fo->Write(buffer);
+    fo->Write(impl->cuts_.cut_values_.ConstHostVector());
```

Comment:
This is cool. Perhaps we should have an option to keep the ellpack pages between runs for things like hyperparameter tuning.

Reply:
Consensus from #4237 was that we would delete cache after each run to prevent undefined behaviours.